src/mlpack/methods/reinforcement_learning/environment/acrobat.hpp

```diff
@@ -0,0 +1,319 @@
+    return -1;
+  };
+  double Sample(const State& state, const Action& action)
+  {
+    State nextState;
+    return Sample(state, action, nextState);
+  }
+  State InitialSample() const
+  {
+    return State((arma::randu<arma::colvec>(4) - 0.5) / 5.0);
+  }
+  /**
+   *
+   * @param state The current State
+   */        
+  bool IsTerminal(const State& state) const
+  {
+    double theta1 = state.Theta1();
+    double theta2 = state.Theta2();
+    return bool ((-cos(theta1)-cos(theta1+theta2)) > 1.0);
+  }
+  /**
+   * @param state Current State
+   * @param torque Torque Applied 
+   */
+  arma::colvec dsdt(arma::colvec state,
+               const double torque)
+  {
+    double m1 = link_mass1;
+    double m2 = link_mass2;
+    double l1 = link_length1;
+    double l2 = link_length2;
+    double lc1 = link_com1;
+    double lc2 = link_com2;
+    double I1 = link_moi;
+    double I2 = link_moi;
+    double g = gravity;
+    double a = torque;
+    double theta1 = state[0];
+    double theta2 = state[1];
+    double dtheta1 = state[2];
+    double dtheta2 = state[3];
+    double d1 = m1 * pow(lc1, 2) + m2 *
+              (pow(l1, 2) + pow(lc2, 2) + 2 * l1 * lc2 * cos(theta2))
+               + I1 + I2;
+    double d2 = m2 * (pow(lc2, 2) + l1 * lc2 * cos(theta2)) + I2;
+    double phi2 = m2 * lc2 * g * cos(theta1 + theta2 - PIE / 2.);
+
+    double phi1 = - m2 * l1 * lc2 * pow(dtheta2, 2) * sin(theta2)
+      - 2 * m2 * l1 * lc2 * dtheta2 * dtheta1 * sin(theta2)
+      + (m1 * lc1 + m2 * l1) * g * cos(theta1 - PIE / 2)
+      + phi2;
+
+    double ddtheta2 = (a + d2 / d1 * phi1 - m2 * l1 * lc2 * pow(dtheta1, 2) *
+     sin(theta2) - phi2) / (m2 * pow(lc2, 2) + I2 - pow(d2, 2) / d1);
+
+    double ddtheta1 = -(d2 * ddtheta2 + phi1) / d1;
+    arma::colvec returnValues = {dtheta1, dtheta2, ddtheta1, ddtheta2};
+    return returnValues;
+  };
+  /**
+   * @param value scalar value to wrap
+   * @param minimum minimum range of wrap
+   * @param maximum maximum range of wrap
+   */      
+  double wrap(double value,
+              double minimum,
+              double maximum)
+  {
+    double diff = maximum - minimum;
+    if (value > maximum) value = value - diff;
+    else if (value < minimum) value = value + diff;
+    return value;
+  };
+  /**
+    * @param value scalar value to bound
+    * @param minimum minimum range of bound
+    * @param maximum maximum range of bound
+    */     
+  double  bound(double value,
+                double minimum,
+                double maximum)
+  {
+    return std::min(std::max(value, minimum), maximum);
+  };
+  /**
+    * @param state_ Current State
+    * @param Action Action Taken
+    * @param nextState nextState 
+    */
+  double Torque(const Action& Action)
+  {
+    // Add noise to the Torque
+    std::default_random_engine generator;
+    std::uniform_real_distribution<double> distribution(-0.1, +0.1);
+    double torque = double(Action - 1) + distribution(generator);
```

Comment:
Do you mind to use https://github.com/mlpack/mlpack/blob/793fe4abe47cf03872e77fd00369d1653b450758/src/mlpack/core/math/random.hpp#L79 here. I think that would make the expression cleaner.